business/workloads.go

```diff
@@ -189,17 +194,36 @@ func (in *WorkloadService) GetPod(namespace, name string) (*models.Pod, error) {
 	return &pod, nil
 }
 
-func (in *WorkloadService) getParsedLogs(namespace, name string, opts *core_v1.PodLogOptions) (*PodLog, error) {
-	podLog, err := in.k8s.GetPodLogs(namespace, name, opts)
+func (in *WorkloadService) getParsedLogs(namespace, name string, opts *LogOptions) (*PodLog, error) {
+	k8sOpts := opts.PodLogOptions
+	podLog, err := in.k8s.GetPodLogs(namespace, name, &k8sOpts)
 
 	if err != nil {
 		return nil, err
 	}
 
 	lines := strings.Split(podLog.Logs, "\n")
 	messages := make([]LogEntry, 0)
+	lineCount := int64(0)
```

Comment:
Maybe I'm missing something but I don't think you want to do this TailLines logic, that starts here, on every call.  In normal cases, when querying to `now` we can just call k8s API normally.   This issue is only when `(sinceTime + duration) < now`.  In that case we can't pass PodLogOptions.TailLines to the k8s API because it will apply the tail to the most recent logs.   Instead, I think we need to:
1) trim logs with time > endTime (i.e. `endTime = sinceTime + duration`)
2) apply TailLines to the remaining logs

Reply:
Would it fix the case if I just set the k8s param for it to nil before the query, and then apply the tailLines directly on the code? From what I see, we need to parse all lines anyway, so we can just stop parsing then we have enough lines (tailLines if set, then durations).

Reply:
Yes, I suppose we could do it in all cases, the only cost is the k8s API returning more lines than it really may have to return, but I guess that is fairly minor.  If you prefer that approach I'm OK with it.